### PR TITLE
#2107: CI: remove builds for unsupported compilers

### LINKED ISSUE #2107: remove `gcc-7` build and reproduce its configurations on different build
**What Needs to be Done?**
We have raised the minimum version requirements for both `gcc` and `clang`.

- [x] Remove obsolete builds
- [x] Remove obsolete badges from README
- [ ] Make sure that CI and [build requirements on wiki](https://github.com/DARMA-tasking/vt/wiki/Build-Requirements#supported-compilers) are matching
- [x] Move the configurations used in the removed builds to some other builds (tracing + coverage, address sanitizer)

### PR DESCRIPTION
**Pull request by cz4rs**
- remove build for `gcc-7` (no longer supported)
- move its configuration to `gcc-11` build
- move LB schema test to `gcc-9` build (to avoid having too many options on `gcc-11`)

fixes #2107

### CODE REVIEW

**Review by lifflander — APPROVED** [review]

File: .github/workflows/macosx-clang-mpich.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: PR tests (clang-8, macosx, mpich)
+name: PR tests (clang-14, macosx, mpich)
```
Comment (PhilMiller) [inline .github/workflows/macosx-clang-mpich.yml:1]: The cache name and maybe other stuff still refer to clang 8.  I don't see what in this file actually set the compiler version, though.

Comment (cz4rs) [reply .github/workflows/macosx-clang-mpich.yml:1]: Updated the version number everywhere.

Compiler version is not set explicitly, we use the default for that image: https://github.com/actions/runner-images/blob/macOS-12/20230318.1/images/macos/macos-12-Readme.md.

File: .github/workflows/build-docs.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
       ARCH: amd64
       UBUNTU: 18.04
       COMPILER_TYPE: gnu
-      COMPILER: gcc-7
+      COMPILER: gcc-8
```
Comment (cz4rs) [inline .github/workflows/build-docs.yml:19]: Switching to the newest version (Ubuntu 22.04 + gcc-12) causes error in the docs generation. I have not investigated it and opted for a minimal bump instead.

This seems to be working fine:
https://github.com/DARMA-tasking/vt/actions/runs/4563477715/jobs/8052013152

File: .github/workflows/macosx-clang-mpich.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@ concurrency:
 
 jobs:
   build:
-    runs-on: macos-latest
+    runs-on: macos-12
```
Comment (cz4rs) [inline .github/workflows/macosx-clang-mpich.yml:18]: Changing `latest` to a pinned version (this was considered preferable for Intel compilers, we can probably use the same approach here).

Comment (cz4rs) [reply .github/workflows/macosx-clang-mpich.yml:18]: AFAIK this should be enough to make the compiler version pinned as well, so we don't have to configure it explicitly elsewhere.

**Review by PhilMiller — APPROVED** [review]

**Pull request merged**